cdci_data_analysis/flask_app/dispatcher_query.py

```diff
@@ -1055,6 +1062,10 @@ def download_products(self):
 
             self.validate_job_id(request_parameters_from_scratch_dir=True)
             file_list = self.args.get('file_list').split(',')
+            for fn in file_list:
+                if "../" in fn:
+                    raise RequestNotAuthorized('No such file')
+                        
```

Comment:
why not including this directly inside the `prepare_download` def ?

Reply:
It may be there of course. I put it here so that it has a meaning of arguments validation

Reply:
sure, but within the other function we avoid double loop and some little code duplication (it can be included within the abspath check)